zio-sbt-ci/src/main/scala/zio/sbt/ZioSbtCiPlugin.scala

```diff
@@ -520,40 +517,48 @@ object ZioSbtCiPlugin extends AutoPlugin {
       val jvmOptions       = Seq("-XX:+PrintCommandLineFlags") ++ ciJvmOptions.value
       val nodeOptions      = ciNodeOptions.value
 
-      val jvmMap = Map(
+      val jvmMap = ListMap(
         "JDK_JAVA_OPTIONS" -> jvmOptions.mkString(" ")
       )
-      val nodeMap: Map[String, String] =
-        if (nodeOptions.nonEmpty) Map("NODE_OPTIONS" -> nodeOptions.mkString(" ")) else Map.empty
-
-      val workflow = yaml
-        .Printer(
-          preserveOrder = true,
-          dropNullKeys = true,
-          splitLines = false,
-          lineBreak = LineBreak.Unix,
-          version = YamlVersion.Auto
+      val nodeMap: ListMap[String, String] =
+        if (nodeOptions.nonEmpty) ListMap("NODE_OPTIONS" -> nodeOptions.mkString(" ")) else ListMap.empty
+
+      val yamlOptions =
+        YamlOptions.default.copy(
+          dropNulls = true,
+          lineBreak = org.yaml.snakeyaml.DumperOptions.LineBreak.UNIX,
+          maxScalarWidth = Some(1024)
         )
-        .pretty(
-          Workflow(
-            name = workflowName,
-            env = jvmMap ++ nodeMap,
-            triggers = Seq(
-              Trigger.WorkflowDispatch(),
-              Trigger.Release(Seq("published")),
-              Trigger.Push(branches = enabledBranches.map(Branch.Named)),
-              Trigger.PullRequest(ignoredBranches = Seq(Branch.Named("gh-pages")))
-            ),
-            jobs =
-              buildJobs ++ lintJobs ++ testJobs ++ updateReadmeJobs ++ reportSuccessful ++ releaseJobs ++ postReleaseJobs
-          ).asJson
+
+      val workflow =
+        Workflow(
+          name = workflowName,
+          env = jvmMap ++ nodeMap,
+          on = Some(
+            Triggers(
+              release = Some(Trigger.Release(Seq(Trigger.ReleaseType.Published))),
+              push = Some(Trigger.Push(branches = enabledBranches.map(Branch.Named))).filter(_.branches.nonEmpty),
+              pullRequest = Some(Trigger.PullRequest(branchesIgnore = Seq(Branch.Named("gh-pages"))))
+            )
+          ),
+          jobs = ListMap(
+            (buildJobs ++ lintJobs ++ testJobs ++ updateReadmeJobs ++ reportSuccessful ++ releaseJobs ++ postReleaseJobs)
+              .map(job => job.id -> job): _*
+          )
         )
 
+      val yaml: String = zio.json.ast.Json.decoder
+        .decodeJson(workflow.toJson)
```

Comment:
this is a bit dirty but the toJsonAst does not perform the `excludeEmptyCollections` , will investigate this later in zio-json